src/Uno.UI.Runtime.Skia.Wpf/Islands/WpfIslandsHost.cs

```diff
@@ -0,0 +1,276 @@
+			}
+		}
+
+		private void Current_Deactivated(object? sender, EventArgs e)
+		{
+			var winUIWindow = WinUI.Window.Current;
+			winUIWindow?.OnActivated(Windows.UI.Core.CoreWindowActivationState.Deactivated);
+		}
+
+		private void Current_Activated(object? sender, EventArgs e)
+		{
+			var winUIWindow = WinUI.Window.Current;
+			winUIWindow?.OnActivated(Windows.UI.Core.CoreWindowActivationState.CodeActivated);
+		}
+
+		private void WpfHost_Loaded(object sender, RoutedEventArgs e)
+		{
+			WinUI.Window.Current.OnNativeSizeChanged(new Windows.Foundation.Size(ActualWidth, ActualHeight));
+		}
+
+		private void WpfHost_SizeChanged(object sender, SizeChangedEventArgs e)
+		{
+			WinUI.Window.Current.OnNativeSizeChanged(
+				new Windows.Foundation.Size(
+					e.NewSize.Width,
+					e.NewSize.Height
+				)
+			);
+		}
+
+		public SKSize CanvasSize => bitmap == null ? SKSize.Empty : new SKSize(bitmap.PixelWidth, bitmap.PixelHeight);
+
+		public bool IgnorePixelScaling
+		{
+			get => ignorePixelScaling;
+			set
+			{
+				ignorePixelScaling = value;
+				InvalidateVisual();
+			}
+		}
+
+		WinUI.XamlRoot? IWpfHost.XamlRoot => throw new NotImplementedException();
+
+		protected override void OnRender(DrawingContext drawingContext)
+		{
+			base.OnRender(drawingContext);
+
+			if (designMode)
+			{
+				return;
+			}
+
+			if (ActualWidth == 0
+				|| ActualHeight == 0
+				|| double.IsNaN(ActualWidth)
+				|| double.IsNaN(ActualHeight)
+				|| double.IsInfinity(ActualWidth)
+				|| double.IsInfinity(ActualHeight)
+				|| Visibility != Visibility.Visible)
+			{
+				return;
+			}
+
+
+			int width, height;
+
+			if (_displayInformation == null)
+			{
+				_displayInformation = DisplayInformation.GetForCurrentView();
+			}
+
+			var dpi = _displayInformation.RawPixelsPerViewPixel;
+			double dpiScaleX = dpi;
+			double dpiScaleY = dpi;
+			if (IgnorePixelScaling)
+			{
+				width = (int)ActualWidth;
+				height = (int)ActualHeight;
+			}
+			else
+			{
+				var matrix = PresentationSource.FromVisual(this).CompositionTarget.TransformToDevice;
+				dpiScaleX = matrix.M11;
+				dpiScaleY = matrix.M22;
+				width = (int)(ActualWidth * dpiScaleX);
+				height = (int)(ActualHeight * dpiScaleY);
+			}
+
+			var info = new SKImageInfo(width, height, SKImageInfo.PlatformColorType, SKAlphaType.Premul);
+
+			// reset the bitmap if the size has changed
+			if (bitmap == null || info.Width != bitmap.PixelWidth || info.Height != bitmap.PixelHeight)
+			{
+				bitmap = new WriteableBitmap(width, height, 96 * dpiScaleX, 96 * dpiScaleY, PixelFormats.Pbgra32, null);
+			}
+
+			// draw on the bitmap
+			bitmap.Lock();
+			using (var surface = SKSurface.Create(info, bitmap.BackBuffer, bitmap.BackBufferStride))
+			{
+				surface.Canvas.Clear(SKColors.White);
+				surface.Canvas.SetMatrix(SKMatrix.CreateScale((float)dpiScaleX, (float)dpiScaleY));
+				WinUI.Window.Current.Compositor.Render(surface);
+			}
```

Comment:
Should we be reusing the same rendering code as in the other host? Duplication will make adding OpenGL harder.